### ISSUE TITLE
LeBlender Umbraco Exception - An item with the same key has already been added 

### ISSUE THREAD

**Issue by blevai13** [Lecoati/LeBlender#35]
We are facing a strange behaviour with LeBlender editor,  we have the following scenario we setup a datatype in backoffice, which is leblender type. Then we pick up a new property which is an archetype datatype and this child could hold as you may know complex datatypes contents as user wants. Now the tricky part is coming, because we set here a grid content page which can holds modules. These modules are all kind of types, doesn't matter, mostly BUT, if we have a module which type is LeBlender then we've got an exception that ,, An item with the same key has already been added" - so basicly the problem is, I think then when leblender render the editor it gives the same key for both - same for the parent and same for the child. Basically we have leblender editor in leblender editor. At backoffice side everything looks great.
Could you point me out what should we change, or maybe you could give us some updates?! Thanks in advance!

**Comment by ArnoldV** [comment]
Yes, we have the same issue, will try to fix it and send a pull requests.

our code that breaks is:
```
@inherits UmbracoViewPage<Lecoati.LeBlender.Extension.Models.LeBlenderModel​>

@{
    var currentPage = Lecoati.LeBlender.Extension.Helper.GetCurrentContent();

    foreach (var child in currentPage.Children.Where(c => !c.GetPropertyValue<bool>("umbracoNaviHide")))
     {
         @Html.GetGridHtml(child, "layoutGrid")
     }
}
```

**Comment by ArnoldV** [comment]
Looking into this it seems te be caused by this line:

https://github.com/Lecoati/LeBlender/blob/d194b15c06c53d6b72d4ceb1deeae7e66334fd4b/Src/Lecoati.LeBlender.Ui/App_Plugins/LeBlender/editors/leblendereditor/views/Base.cshtml#L14

if you remove the 'this.ViewData' from the creation of the new ViewDataDictionary things work.
It might be that I am overlooking something though that will now break?

**Comment by ArnoldV** [comment]
So basically this commit should be reverted: https://github.com/Lecoati/LeBlender/commit/3516de30424ecea47ddb094c663b1b1a8078d2ea

**Comment by Bitmapped** [comment]
This revert causes problems for me because properties I add to ViewData from my _ViewStart file are now lost. This needs to go back to the drawing board for a better fix.
